server/middlewares/checkKeyword.js

```diff
@@ -0,0 +1,26 @@
+/**
+  * This middleware intercepts the request and checks that the request
+  * contains a search keyword in the query parameter.
+  * @param {object} req the request object
+  * @param {object} res the response object
+  * @param {object} next the next middleware function
+
+  */
+const checkKeyword = (req, res, next) => {
+  const keyword = req.query.q;
+  if (keyword === undefined) {
+    return res.status(400).json({
+      status: 'fail',
+      message: 'Please enter a search keyword'
+    });
+  }
+  if (keyword === '' || keyword == null) {
```

Comment:
I feel you could have added these checks in the `if` statement on line 11, instead of having multiple ifs that basically return the same result.

Reply:
Thanks for this review, I will do the needful to the codebase.